src/libcollections/linked_list.rs

```diff
@@ -1020,8 +1020,8 @@ impl<I: IntoIterator> SpecExtend<I> for LinkedList<I::Item> {
 }
 
 impl<T> SpecExtend<LinkedList<T>> for LinkedList<T> {
-    fn spec_extend(&mut self, ref mut other: LinkedList<T>) {
-        self.append(other);
+    fn spec_extend(&mut self, mut other: LinkedList<T>) {
```

Comment:
Rust doesn't implicitly copy things unless they are Copy, and heap-allocating collections are never Copy. Typical types that implement Copy are `i32` and `&str`.